service/src/misc.rs

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+use crate::error::ServiceError;
+use crate::error::ServiceError::*;
+use crate::storage_impl::{get_data, SQLiteStorage};
+use crate::{TRUSTED_TIMESTAMP_FUNCTION_NAME, TRUSTED_TIMESTAMP_SERVICE_ID};
+use fluence_keypair::PublicKey;
+use libp2p_core::PeerId;
+use marine_rs_sdk::CallParameters;
+use parking_lot::MutexGuard;
+use std::convert::TryFrom;
+use std::str::FromStr;
+use trust_graph::TrustGraph;
+
+/// Check timestamps are generated on the current host with builtin ("peer" "timestamp_sec")
+pub(crate) fn check_timestamp_tetraplets(
+    call_parameters: &CallParameters,
+    arg_number: usize,
+) -> Result<(), ServiceError> {
+    let tetraplets = call_parameters
+        .tetraplets
+        .get(arg_number)
+        .ok_or(InvalidTimestampTetraplet)?;
+    let tetraplet = tetraplets.get(0).ok_or(InvalidTimestampTetraplet)?;
+    (tetraplet.service_id == TRUSTED_TIMESTAMP_SERVICE_ID
+        && tetraplet.function_name == TRUSTED_TIMESTAMP_FUNCTION_NAME
```

Comment:
If `TRUSTED_TIMESTAMP = ("peer", "timestamp_sec")`
```suggestion
    ((tetraplet.service_id, tetraplet.function_name) == TRUSTED_TIMESTAMP
```